src/libcore/alloc.rs

```diff
@@ -140,6 +140,36 @@ impl Layout {
         unsafe { Layout::from_size_align_unchecked(size, align) }
     }
 
+    /// Produces layout describing a record that could be used to
+    /// allocate backing structure for `T` (which could be a trait
+    /// or other unsized type like a slice).
+    ///
+    /// # Safety
+    ///
+    /// This function is only safe to call if one of the following conditions hold:
+    ///
+    /// - If `T` is `Sized`, this function is always safe to call.
+    /// - If the unsized tail of `T` is:
+    ///     - a [slice], then the length of the slice tail must be an intialized
+    ///       integer, and the size of the described `T` must fit in `isize`.
+    ///     - a [trait object], then the vtable part of the pointer must point
+    ///       to a valid vtable acquired by an unsizing coersion.
+    ///     - otherwise, it is conservatively not allowed to call this function.
+    ///
+    /// This restriction may be loosened in the future.
```

Comment:
I don't entirely see the point of emphasizing this, isn't that basically implicit anywhere?